Add GOVERNANCE.rst, based on the WarpX model (#3917)
AMReX-Codes/amrex: GOVERNANCE.rst

SC members can call votes for decisions (technical or governance).

The SC can veto decisions of the technical committee (TC) by voting in the SC.
The TC can overwrite a veto with a 2/3rd majority vote in the TC.
Decisions are documented in developer meeting notes and/or on the GitHub repository.

The SC can change the governance structure, but only in a unanimous vote.

Decision Process
^^^^^^^^^^^^^^^^

Decision of the SC usually happen in the developer meetings, via e-mail or public chat.

Decisions are made in a non-confidential manner, by majority on the cast votes of SC members.
Votes can be cast in asynchronous manner, e.g., over the time of 1-2 weeks.
In tie situations, the chair of the SC acts as the tie breaker.

Appointment Process
^^^^^^^^^^^^^^^^^^^

Appointed by current SC members in an unanimous vote.
As a SC member, regularly attending and contributing to the weekly developer meetings is expected.

SC members can resign or be removed by majority vote, e.g., due to inactivity, bad acting or other reasons.

Technical Committee
-------------------

Current Roster
^^^^^^^^^^^^^^

- Axel Huebl

> - Kevin Gott

Reviewer: Only suggestion: as in WarpX, you might want to include SC members that do day-to-day technical work in the TC as well.

Author: Good call.

- Marc Day
- Michael Zingale

> See: `GitHub team <https://github.com/orgs/AMReX-Codes/teams/amrex>`__

Author: Note that we need to tweak the settings for the AMReX-Codes organization to match the three-team structure in this document.

.. _governance:

AMReX Governance
================

AMReX is led in an open governance model, described in this file.

Steering Committee
------------------

Current Roster
^^^^^^^^^^^^^^

- Ann Almgren
- John Bell (chair)
- Andrew Myers
- Weiqun Zhang

Role
^^^^

Members of the steering committee (SC) can change organizational settings, do administrative operations such as rename/move/archive repositories, change branch protection rules, etc.

> SC members can call votes for decisions (technical or governance).

Reviewer 2: add a blank line before this line

The SC can veto decisions of the technical committee (TC) by voting in the SC.

> The TC can overwrite a veto with a 2/3rd majority vote in the TC.

Reviewer 3: "overwrite" should be "override"

Decisions are documented in developer meeting notes and/or on the GitHub repository.

The SC can change the governance structure, but only in a unanimous vote.

Decision Process
^^^^^^^^^^^^^^^^

> Decision of the SC usually happen in the developer meetings, via e-mail or public chat.

Reviewer 3: "Decision" --> "Decisions"

> Decisions are made in a non-confidential manner, by majority on the cast votes of SC members.

Reviewer 3: "majority on" -- "majority of"

> Decisions are made in a non-confidential manner, by majority on the cast votes of SC members.

Reviewer 2: change after comma to "by the majority of the votes cast by SC members."  add a blank line after this

> Votes can be cast in asynchronous manner, e.g., over the time of 1-2 weeks.

Reviewer 3: "Votes can be cast asynchronously, e.g., over a time period of 1-2 weeks."

In tie situations, the chair of the SC acts as the tie breaker.

Appointment Process
^^^^^^^^^^^^^^^^^^^

> Appointed by current SC members in an unanimous vote.

Reviewer 3: Replace by "New members of the SC can be appointed by unanimous vote of the current SC members"

> As a SC member, regularly attending and contributing to the weekly developer meetings is expected.

Reviewer 3: Replace by "SC members are expected to attend and contribute to regular developer meetings."

> SC members can resign or be removed by majority vote, e.g., due to inactivity, bad acting or other reasons.

Reviewer 3: "bad acting" --> "inappropriate or otherwise negative behavior"

Technical Committee
-------------------

Current Roster
^^^^^^^^^^^^^^

- Ann Almgren
- Marc Day
- Candace Gilet
- Kevin Gott
- Axel Huebl
- Andrew Myers
- Weiqun Zhang
- Michael Zingale

See: `GitHub team <https://github.com/orgs/AMReX-Codes/teams/amrex>`__

Role
^^^^

> The technical committee (TC) is the core governance body, where under normal operations most ideas are discussed and decisions are made.

Reviewer 3: "where under normal operations most" --> "where, under normal operations, most"

Individual TC members can approve and merge code changes.

> Usually, they seek approval by another maintainer for their own changes, too.

Reviewer 3: "TC members are expected to seek approval of another maintainer for their own changes, except under exigent circumstances."

> TC members lead - and weigh in on - technical discussions and, if needed, can call for a vote between TC members for a technical decision.

Reviewer 3: "vote between TC members" --> "vote of the TC"

> TC members merge/close PRs and issues, and moderate (including block/mute) bad actors.

Reviewer 3: "TC members merge/close" --> "TC members can merge/close"

The TC can propose governance changes to the SC.

Decision Process
^^^^^^^^^^^^^^^^

Discussion in the TC usually happens in the developer meetings. Developer meetings can be scheduled by any member of the SC or TC.

> If someone calls for a vote to make a decision: majority based on the cast votes; we need 50% of the committee participating to vote. In the absence of a quorum, the SC will decide according to its voting rules.

Reviewer 3: Convert this to "If a member of the TC calls for a vote, the vote will be decided by the majority of the votest cast, provided that at least half of the TC members participate in the vote.   If fewer than half of the TC members cast votes, the SC will make the decision according to the guidelines specified above."

Votes are cast in a non-confidential manner.

> Decisions are documented in the developer meeting notes and/or on the GitHub repository.

Reviewer 3: "in the GitHub repository" instead of "on the GitHub repository"??

> TC members can individually appoint new contributors, unless a vote is called on an individual.

Reviewer 3: I think we should require another TC member to second the suggestion of a new member.  So change this line to read "Individual TC members can suggest the addition of new contributors.   The suggestion must be seconded by another TC member.  Any TC member has the right to oppose the suggestion and call for a vote on the decision."

Appointment Process
^^^^^^^^^^^^^^^^^^^

TC members are the maintainers of AMReX.

> As a TC member, regularly attending and contributing to the developer meetings is expected.

Reviewer 2: TC members are expected to attend and contribute to regular developer meetings.

> One is appointed to the TC by the steering committee, in a unanimous vote, or by majority vote of the TC. The SC can veto appointments.

Reviewer 3: "New TC members can be suggested by either SC or TC members.   Confirmation of a new TC member requires a majority of votes cast by either the SC or the TC.   The SC can veto any new appointment."

Steering committee members can also be TC members.

> TC members can resign or be removed by majority vote by either TC or SC, e.g., due to inactivity, bad acting or other reasons.

Reviewer 3: make same change to "bad acting" as in comment above

Contributors
------------

Current Roster
^^^^^^^^^^^^^^

See: `GitHub contributors <https://github.com/AMReX-Codes/amrex/graphs/contributors>`__

Role
^^^^

Contributors are valuable, vetted developers of AMReX.
Contributions can be in many forms and not all need to be code contributions.
Examples include code pull requests, support in issues & user discussions, writing and updating documentation, writing tutorials, visualizations, R&D on algorithms, testing and benchmarking, etc.
Contributors can participate in developer meetings and weigh in on discussions.

> Contributors can "triage" (add labels) to pull requests, issues, and GitHub discussion pages.

Reviewer 3: remove "to" in "to pull requests"

> Contributors can comment and review PRs (but not merge).

Reviewer 3: "comment" --> "comment on"

Decision Process
^^^^^^^^^^^^^^^^

Contributors can individually decide on classification (triage) of pull requests, issues, and GitHub discussion pages.

Appointment Process
^^^^^^^^^^^^^^^^^^^

> Appointed after contributing to AMReX (see above) by any member of the TC.

Reviewer 3: Let's also add that this must be seconded by another member of the TC.

> The role can be lost by resigning or by decision of an individual TC or SC member, e.g., due to inactivity, bad acting or other.

Reviewer 3: Let's make this majority vote like the other cases above where we make someone leave.  Change "bad acting" as before

Former Members
--------------

> "Former members" are *not* tracked as a governance role in AMReX.

Reviewer 3: "Former contributors do not play any role in the governance of AMReX."

Instead, former (e.g., inactive) contributors are acknowledged separately in GitHub contributor tracking, the AMReX documentation, references, etc. as appropriate.

> Former members of SC, TC and Contributors are not kept in the roster, since committee role rosters should reflect currently active members and the responsible governance body.

Reviewer 3: Former members of the SC or TC also do not play any role in the governance of AMReX.

.. _governance:

AMReX Governance
================

> AMReX is led in an open governance model, described in this file.

Reviewer 4: `file` -> `document`?
```suggestion
AMReX is led in an open governance model, described in this document.
```

Decisions are made in a non-confidential manner, by the majority of votes cast by SC members.

Votes can be cast asynchronously, e.g., over a time period of 1-2 weeks.
In tie situations, the chair of the SC acts as the tie breaker.

Appointment Process
^^^^^^^^^^^^^^^^^^^

New members of the SC can be appointed by unanimous vote of the current SC members.

SC members are expected to attend and contribute to regular developer meetings.

SC members can resign or be removed by majority vote, e.g., due to inactivity, inappropriate or otherwise negative behavior, or other reasons.

Technical Committee
-------------------

Current Roster
^^^^^^^^^^^^^^

- Ann Almgren
- Marc Day
- Candace Gilet
- Kevin Gott
- Axel Huebl
- Andrew Myers
- Weiqun Zhang
- Michael Zingale

See: `GitHub team <https://github.com/orgs/AMReX-Codes/teams/amrex>`__

Role
^^^^

The technical committee (TC) is the core governance body, where, under normal operations, most ideas are discussed and decisions are made.
Individual TC members can approve and merge code changes.
TC members are expected to seek approval of another maintainer for their own changes, except under exigent circumstances.

> TC members lead - and weigh in on - technical discussions and, if needed, can call for a vote of the TC for a technical decision.

Reviewer 4: Should `-` be `--`?

Reviewer 3: yeah probably :-)

Reviewer: Or simple
```suggestion
TC members lead and weigh in on technical discussions and, if needed, can call for a vote of the TC for a technical decision.
```

> TC members can merge/close PRs and issues, and moderate (including block/mute) bad actors.

Reviewer 4: Use gerund forms? including blocking/muting?

Reviewer 3: yes -- you are correct -- "(including blocking or muting)"
```suggestion
TC members can merge/close PRs and issues, and moderate (including blocking or muting) bad actors.
```

The TC can propose governance changes to the SC.

Decision Process
^^^^^^^^^^^^^^^^

Discussion in the TC usually happens in the developer meetings. Developer meetings can be scheduled by any member of the SC or TC.

> If a member of the TC calls for a vote, the vote will be decided by the majority of the votest cast, provided that at least half of the TC members participate in the vote. If fewer than half of the TC members cast votes, the SC will make the decision according to the guidelines specified above.

Reviewer 3: "votest" --> "votes"
```suggestion
If a member of the TC calls for a vote, the vote will be decided by the majority of the votes cast, provided that at least half of the TC members participate in the vote. If fewer than half of the TC members cast votes, the SC will make the decision according to the guidelines specified above.
```

Appointment Process
^^^^^^^^^^^^^^^^^^^

TC members are the maintainers of AMReX.
TC members are expected to attend and contribute to regular developer meetings.

New TC members can be suggested by either SC or TC members. Confirmation of a new TC member requires a majority of votes cast by either the SC or the TC. The SC can veto any new appointment.

Steering committee members can also be TC members.

TC members can resign or be removed by majority vote by either TC or SC due to inactivity, inappropriate or otherwise negative behavior, or other reasons.

Contributors
------------

Current Roster
^^^^^^^^^^^^^^

See: `GitHub contributors <https://github.com/AMReX-Codes/amrex/graphs/contributors>`__

Role
^^^^

Contributors are valuable, vetted developers of AMReX.
Contributions can be in many forms and not all need to be code contributions.
Examples include code pull requests, support in issues & user discussions, writing and updating documentation, writing tutorials, visualizations, R&D on algorithms, testing and benchmarking, etc.
Contributors can participate in developer meetings and weigh in on discussions.

> Contributors can "triage" (i.e. add labels to) pull requests, issues, and GitHub discussion pages.

Reviewer 4: Add a comma after i.e.?
```suggestion
Contributors can "triage" (i.e., add and remove labels to) pull requests, issues, and GitHub discussion pages.
```

Contributors can comment on and review PRs (but not merge).

Decision Process
^^^^^^^^^^^^^^^^

Contributors can individually decide on classification (triage) of pull requests, issues, and GitHub discussion pages.

Appointment Process
^^^^^^^^^^^^^^^^^^^

Appointed after contributing to AMReX (see above) through nomination by any member of the TC. Another member of the TC must second the nomination.

> The role can be lost by resigning or by decision of an individual TC or SC member due to inactivity, inappropriate or otherwise negative behavior, or other reasons.

Author: ```suggestion
The role can be lost by resigning or by majority vote of either the TC or the SC due to inactivity, inappropriate or otherwise negative behavior, or other reasons.
```

The SC can veto decisions of the technical committee (TC) by voting in the SC.
The TC can override a veto with a 2/3rd majority vote in the TC.
Decisions are documented in developer meeting notes and/or on the GitHub repository.

The SC can change the governance structure, but only in a unanimous vote.

Decision Process
^^^^^^^^^^^^^^^^

Decisions of the SC usually happen in the developer meetings, via e-mail or public chat.

Decisions are made in a non-confidential manner, by the majority of votes cast by SC members.

Votes can be cast asynchronously, e.g., over a time period of 1-2 weeks.
In tie situations, the chair of the SC acts as the tie breaker.

Appointment Process
^^^^^^^^^^^^^^^^^^^

New members of the SC can be appointed by unanimous vote of the current SC members.

SC members are expected to attend and contribute to regular developer meetings.

SC members can resign or be removed by majority vote, e.g., due to inactivity, inappropriate or otherwise negative behavior, or other reasons.

Technical Committee
-------------------

Current Roster
^^^^^^^^^^^^^^

- Ann Almgren
- Marc Day
- Candace Gilet
- Kevin Gott
- Axel Huebl
- Andrew Myers

> - Weiqun Zhang

Author: ```suggestion
- Andy Nonaka
- Jean Sexton
- Weiqun Zhang
```
